osu.Game/Beatmaps/Formats/LegacyDecoder.cs

```diff
@@ -226,12 +226,16 @@ internal class LegacySampleControlPoint : SampleControlPoint, IEquatable<LegacyS
 
             public override HitSampleInfo ApplyTo(HitSampleInfo hitSampleInfo)
             {
-                var baseInfo = base.ApplyTo(hitSampleInfo);
-
-                if (baseInfo is ConvertHitObjectParser.LegacyHitSampleInfo legacy && legacy.CustomSampleBank == 0)
-                    return legacy.With(newCustomSampleBank: CustomSampleBank);
+                if (hitSampleInfo is ConvertHitObjectParser.LegacyHitSampleInfo legacy)
+                {
+                    return legacy.With(
+                        newCustomSampleBank: legacy.CustomSampleBank > 0 ? legacy.CustomSampleBank : CustomSampleBank,
+                        newVolume: hitSampleInfo.Volume > 0 ? hitSampleInfo.Volume : SampleVolume,
```

Comment:
Probably good to add a `VolumeSpecified` property as well and use that for the sample control point override instead of checking if its greater than zero.

Reply:
Hmm, given what you found above I don't think this is necessary as it's a strictly legacy flow.